### DISCUSSION TITLE
Problem with lamella mesh due to area close to zero

### DISCUSSION THREAD

**Discussion by mauroarcidiacono** [neperfepx/neper#640] [Q&A]
Hi,

I generated and imported a 2D part without mesh to Abaqus (I meshed the part using Abaqus). I am having problems with the simulation because Neper is generating zones where the area is close to zero. The resulting elements in those areas are extremely distorted and locally refining the mesh does not solve the problem. 

**Is there a way to make Neper avoid creating lamella in the second level tessellation that have areas so close to zero?** If possible, I would like to discard those small areas. 

![ti64-part-tess-problem](https://user-images.githubusercontent.com/76496290/211557023-53e0a010-56f0-4f07-89b8-8c566956afb0.png)

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.4.2-16
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 16 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/mauroar/.neperrc').
Info   : ---------------------------------------------------------------

```
neper -T -n 175 -domain "square(0.6,0.1)" -dim 2 -id 6 -morpho "diameq:lognormal(1,0.35),1-sphericity:lognormal(0.145,0.03)" -reg 1 -oridescriptor euler-bunge -format tess -o polycrystal_basic

neper -M polycrystal_basic.tess -elttype hex -statelset area -order 1 -rcl 1 -meshqualexpr Odis -meshqualmin 1 -format inp,msh -o polycrystal_basic

neper -T -n 175::from_morpho -domain "square(0.6,0.1)" -dim 2 -id 6::1 -morpho "gg::lamellar(w=file(lamwidth.txt),v=file(lamnormal.txt),pos=optimal)" -reg 1 -rsel 0.1 -oridescriptor euler-bunge -group lam -statcell "scaleid(1)" -format tess,ori -o polycrystal_laminated

neper -V polycrystal_laminated.tess -datacellcol id -datacelltrs 0.7 -print polycrystal_laminated_image

neper -M polycrystal_laminated.tess -elttype hex -statelset area,eltnb -order 1 -rcl 0.1 -meshqualexpr Odis -meshqualmin 1 -format inp,msh -o polycrystal_laminated

neper -V polycrystal_laminated.msh -dataelsetcol id -print mesh
```

[lamnormal.txt](https://github.com/neperfepx/neper/files/10382889/lamnormal.txt)
[lamwidth.txt](https://github.com/neperfepx/neper/files/10382890/lamwidth.txt)

Thank you very much for your help, I really appreciate it.

Kind regards,
Mauro Francisco Arcidiacono

**Comment by rquey** [comment]
`optimal` does not seem to work well in your case. If you want to modify the code, see [here](https://github.com/neperfepx/neper/blob/9047f6e25137bbc12075f7a9f75a3ef1ae175e9e/src/neper_t/net_tess/net_tess_lam/net_tess_lam_seed/net_tess_lam_seed4.c). Unfortunately, I can't prioritize it for now.
